src/main/java/io/pivotal/cfapp/service/ResourceMetadataService.java

```diff
@@ -34,11 +35,52 @@ public ResourceMetadataService(
         this.tokenProvider = tokenProvider;
         this.settings = settings;
     }
-
     private Mono<String> getOauthToken() {
         tokenProvider.invalidate(connectionContext);
         return tokenProvider.getToken(connectionContext);
     }
+    public Mono<Resources> getResources(String type) {
+        ResourceType rt = ResourceType.from(type);
+        final String uri =
+            UriComponentsBuilder
+                .newInstance()
+                    .scheme("https")
+                    .host(settings.getApiHost())
+                    .path("/v3/{type}")
+                    .buildAndExpand(rt.getId())
+                    .encode()
+                    .toUriString();
+        return
+            getOauthToken()
+                .flatMap(t -> webClient
+                                .get()
+                                    .uri(uri)
+                                    .header(HttpHeaders.AUTHORIZATION, t)
+                                        .retrieve()
+                                            .bodyToMono(Resources.class));
+    }
+
+
+    public Mono<Resources> getResources(String type, String labelSelector) {
+        ResourceType rt = ResourceType.from(type);
+        final String uri =
+            UriComponentsBuilder
+                .newInstance()
+                    .scheme("https")
+                    .host(settings.getApiHost())
+                    .path("/v3/{type}")
+                    .queryParam("label_selector", labelSelector)
```

Comment:
This is the crux of what you're wanting to return to the interested party (org owner)? It's the resource(s) that match(es) the label selector?